File: solutions/elixir/darts/2/lib/darts.ex
import :math

defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score(position) do
    distance = calc_distance(position)
    cond do
      distance > 10 -> 0
      distance > 5 -> 1
      distance > 1 -> 5
      true -> 10
    end
  end

  def calc_distance({x, y}) do
    :math.sqrt(x*x + y*y)
  end
end

File: solutions/lua/darts/1/darts.lua
local Darts = {}

function Darts.score(x, y)
  local distance = math.sqrt(x^2 + y^2)
  local score = 0

  if distance > 10 then
    score = 0
  elseif distance > 5 then
    score = 1
  elseif distance > 1 then
    score = 5
  else
    score = 10
  end

  return score
end

return Darts
